pod.go

```diff
@@ -727,6 +727,28 @@ func (p *Pod) findContainer(containerID string) (*Container, error) {
 		containerID, p.id)
 }
 
+// removeContainer removes a container from the containers list held by the
+// pod structure, based on a container ID.
+func (p *Pod) removeContainer(containerID string) error {
+	if p == nil {
```

Comment:
Nit: Great to see lots of error checking, but I don't think this scenario is possible?

Reply:
Yeah you're right, let me remove this from this function and from the other one `findContainer()`

Reply:
Actually, this is possible... The tests are testing this possibility and there is no segfault. So yes basically it could happen. Let's keep the extra check then !